### PR TITLE
Swift Rewrite

### PR DESCRIPTION
**Pull request by justincampbell**

### CODE REVIEW

**Comment by justincampbell** [comment]
Random thoughts:
- Should `IconOuput` return all 3 (ok, warn, error) emoji, or only the relevant emoji?
- Instead of arguments, maybe this should take a format string, something like `%p() %n(q)` and this could also support a time or date format, such as`%t(hh:mm:ss)`.

File: main.swift [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    case .Error:
+      stdout.append(output.errorIcon)
+      if let suffix = output.suffix { stdout.append(suffix) }
+    case .Warn:
+      stdout.append(output.warnIcon)
+      if let suffix = output.suffix { stdout.append(suffix) }
+    case .Ok:
+      if !options.quiet {
+        stdout.append(output.okIcon)
+        if let suffix = output.suffix { stdout.append(suffix) }
```
Comment (justincampbell) [inline main.swift]: Would be nice to remove the duplication for this line:

``` swift
if let suffix = output.suffix { stdout.append(suffix) }
```

**Comment by apbendi** [comment]
hey @justincampbell, took a look at this- maybe you can spec out a little, and/or write tests, for what results you're going for? Is the goal just to reproduce the exact functions of the bash version? I thought you'd mentioned wanting the `q` argument to do something different based on where it was positioned, or no?

**Comment by justincampbell** [comment]
@apbendi Yeah I think a better format long term is to have a format string. I don't know what that should look like. It should support arbitrary strings (spaces) in between elements, and also support quieting each individually. So, if I wanted power always shown, and network quiet, and I wanted a space before, between, and after:

```
tmux-status-bar " {p} {nq} "
```

But maybe there's already and existing library/pattern that would do something close enough to that. In Ruby:

```
tmux-status-bar " #{power} #{quiet(network)} "
```

It would be kinda gross `eval`ing that, but it would work. What do you think?

**Comment by apbendi** [comment]
The eval approach is definitely not going to work in Swift, being statically typed. Leaving aside the ability to format it arbitrarily, you could perhaps used cased arguments to allow varying quietness on each parameter (i.e. -pN would be power, quiet network).

That said, if the grammar has two keywords, it should be pretty straightforward to write a simple parser, and that could be fun too!

**Comment by justincampbell** [comment]
@apbendi Yeah, flags are definitely easier, and we can just decide on single space in between, and let the user wrap in spaces if needed. I think lowercase/uppsercase args is a good solution for quiet-ness.

**Comment by justincampbell** [comment]
@apbendi Also, take another look, I just pushed my WIP from last week. It's probably worth starting an Xcode project from scratch and using this code as a proof of concept.

**Comment by justincampbell** [comment]
I started an Xcode project, and pushed it to master: https://github.com/justincampbell/terminal-status-bar

**Pull request closed without merging**